internal/metricdata/nfstatus.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package metricdata
+
+import (
+	"github.com/omec-project/metricfunc/internal/promclient"
+	"github.com/omec-project/metricfunc/pkg/metricinfo"
+)
+
+func GetNfStatusbyNfType(nfType string) []metricinfo.CNfStatus {
+	var nfs []metricinfo.CNfStatus
+	metricData.NfStatusLock.RLock()
+	defer metricData.NfStatusLock.RUnlock()
+
+	for _, nfStatus := range metricData.NfStatus {
+		if nfStatus.NfType == metricinfo.NfType(nfType) {
+			nfs = append(nfs, *nfStatus)
+		}
+	}
+	return nfs
+}
+
+func GetNfStatusAll() []metricinfo.CNfStatus {
+	var nfs []metricinfo.CNfStatus
+	metricData.NfStatusLock.RLock()
+	defer metricData.NfStatusLock.RUnlock()
+
+	for _, nfStatus := range metricData.NfStatus {
+		nfs = append(nfs, *nfStatus)
+	}
+	return nfs
+}
+
+func HandleNfStatusEvent(nfStatus *metricinfo.CNfStatus) {
+	metricData.NfStatusLock.Lock()
+	defer metricData.NfStatusLock.Unlock()
+
+	metricData.NfStatus[nfStatus.NfName] = nfStatus
```

Comment:
We can discuss this. checking state will require --> read op to see if entry exist or not and then another read to compare state change. This will just save call to update prometheus client.